slatedb/src/flush.rs

```diff
@@ -31,4 +32,226 @@ impl DbInner {
+        ],
+    })]
+    #[case::flush_all_seqs(FlushImmTableTestCase {
+        min_active_seq: 0,
+        row_entries: vec![
+            RowEntry::new_value(&Bytes::from("key"), b"value1", 1),
+            RowEntry::new_value(&Bytes::from("key"), b"value3", 3),
+            RowEntry::new_value(&Bytes::from("key"), b"value2", 2),
+        ],
+        expected_entries: vec![
+            (Bytes::from("key"), 3, ValueDeletable::Value(Bytes::from("value3"))),
+            (Bytes::from("key"), 2, ValueDeletable::Value(Bytes::from("value2"))),
+            (Bytes::from("key"), 1, ValueDeletable::Value(Bytes::from("value1"))),
+        ],
+    })]
+    #[case::flush_some_highest_seqs(FlushImmTableTestCase {
+        min_active_seq: 2,
+        row_entries: vec![
+            RowEntry::new_value(&Bytes::from("key"), b"value1", 1),
+            RowEntry::new_value(&Bytes::from("key"), b"value3", 3),
+            RowEntry::new_value(&Bytes::from("key"), b"value2", 2),
+        ],
+        expected_entries: vec![
+            (Bytes::from("key"), 3, ValueDeletable::Value(Bytes::from("value3"))),
+            (Bytes::from("key"), 2, ValueDeletable::Value(Bytes::from("value2"))),
+        ],
+    })]
+    #[case::flush_only_highest_seq(FlushImmTableTestCase {
+        min_active_seq: 3,
+        row_entries: vec![
+            RowEntry::new_value(&Bytes::from("key"), b"value1", 1),
+            RowEntry::new_value(&Bytes::from("key"), b"value3", 3),
+            RowEntry::new_value(&Bytes::from("key"), b"value2", 2),
+        ],
+        expected_entries: vec![
+            (Bytes::from("key"), 3, ValueDeletable::Value(Bytes::from("value3")))
+        ],
+    })]
+    #[case::flush_highest_seqs_multiple_key(FlushImmTableTestCase {
+        min_active_seq: 6,
+        row_entries: vec![
+            RowEntry::new_value(&Bytes::from("key1"), b"value1", 1),
+            RowEntry::new_value(&Bytes::from("key1"), b"value2", 2),
+            RowEntry::new_value(&Bytes::from("key2"), b"value3", 3),
+            RowEntry::new_value(&Bytes::from("key3"), b"value4", 4),
+            RowEntry::new_value(&Bytes::from("key1"), b"value5", 5),
+            RowEntry::new_value(&Bytes::from("key2"), b"value6", 6),
+        ],
+        expected_entries: vec![
+            // This is the expected results, because for each key slate needs to
+            // a value at or before the min_active_seq
+            // (see retention_iterator for more details)
+            (Bytes::from("key1"), 5, ValueDeletable::Value(Bytes::from("value5"))),
+            (Bytes::from("key2"), 6, ValueDeletable::Value(Bytes::from("value6"))),
+            (Bytes::from("key3"), 4, ValueDeletable::Value(Bytes::from("value4"))),
+        ],
+    })]
+    #[case::flush_tombstones(FlushImmTableTestCase {
+        min_active_seq: 5,
+        row_entries: vec![
+            RowEntry::new_value(&Bytes::from("key1"), b"value1", 1),
+            RowEntry::new_tombstone(&Bytes::from("key1"), 2),
```

Comment:
can we add multiple tombstones for one key and make sure that works as well? I think either flushing all of them or just flushing the latest one